### PR TITLE
Integrate hyper_serde into Servo source

### PR DESCRIPTION
**Pull request by mrobinson**
hyper_serde changes very infrequently and typically only in order to
upgrade Servo dependencies. crates.io lists hyper_serde as having no
dependents, and its integration will not prevent releasing new versions.
Integrating it will both reduce the number of repositories we have to
maintain as well as making it easier to upgrade Servo dependencies.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they do not change behavior.

### CODE REVIEW

File: components/hyper_serde/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "hyper_serde"
+version = "0.13.2"
+edition = "2018"
+authors = ["Anthony Ramine <n.oxyde@gmail.com>"]
+description = "Serde support for hyper types."
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/servo/hyper_serde"
```
Comment (sagudev) [inline components/hyper_serde/Cargo.toml:8]: This should probably change right?

**Comment by mukilan** [comment]
I'm wondering this crate should be in `support/` instead of `components/`. Seems like `components/` should only contain the crates providing browser-engine specific functionality, but this doesn't seem to be followed strictly e.g crates like `range` and `rand` are also in `components/` while `rust-task-info` is in `support/`.

**Comment by sagudev** [comment]
So components should have all necessary crates for compilation, while support optional supporting stuff (crown, some scripts). It would still be nice to have folder inside components that have non-browser crates (servo_arc, hyper, rand, ...).

File: components/hyper_serde/Cargo.toml (hunk 1 above)
Comment (mrobinson) [reply components/hyper_serde/Cargo.toml]: Oh yep. This is fixed in the latest version of the branch.

**Review by sagudev — APPROVED** [review]

**Pull request merged**